java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -13,227 +13,286 @@
+}
 
 /*
  * Class:     ai_onnxruntime_OrtSession
  * Method:    getNumInputs
  * Signature: (JJ)J
  */
-JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_getNumInputs
-  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    size_t numInputs;
-    checkOrtStatus(jniEnv,api,api->SessionGetInputCount((OrtSession*)handle, &numInputs));
-    return numInputs;
+JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_getNumInputs(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  size_t numInputs = 0;
+  checkOrtStatus(jniEnv, api, api->SessionGetInputCount((OrtSession*)handle, &numInputs));
+  return numInputs;
 }
 
 /*
  * Class:     ai_onnxruntime_OrtSession
  * Method:    getInputNames
  * Signature: (JJJ)[Ljava/lang/String;
  */
-JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getInputNames
-  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
-
-    // Setup
-    char *stringClassName = "java/lang/String";
-    jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName);
-
-    // Get the number of inputs
-    size_t numInputs = Java_ai_onnxruntime_OrtSession_getNumInputs(jniEnv, jobj, apiHandle, sessionHandle);
-
-    // Allocate the return array
-    jobjectArray array = (*jniEnv)->NewObjectArray(jniEnv,safecast_size_t_to_jsize(numInputs),stringClazz,NULL);
-    for (uint32_t i = 0; i < numInputs; i++) {
-        // Read out the input name and convert it to a java.lang.String
-        char* inputName;
-        checkOrtStatus(jniEnv,api,api->SessionGetInputName((OrtSession*)sessionHandle, i, allocator, &inputName));
-        jstring name = (*jniEnv)->NewStringUTF(jniEnv,inputName);
-        (*jniEnv)->SetObjectArrayElement(jniEnv, array, i, name);
-        checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,inputName));
+JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getInputNames(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtAllocator* allocator = (OrtAllocator*)allocatorHandle;
+  OrtSession* session = (OrtSession*)sessionHandle;
+
+  // Setup
+  char* stringClassName = "java/lang/String";
+  jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName);
```

Comment:
> jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName); [](http://example.com/codeflow?start=1&length=68)

Can this FindClass() be executed only once?

Reply:
I don't think so, as Java could move the class object on the heap during GC. There is probably a way to pin it, but then I'd have additional native objects in native memory, and the pinning might interfere with the GC compaction.